### PR TITLE
Update `repair` command to allow migration of annotation values

### PR DESCRIPTION
**Pull request by balhoff**
See #492.

Syntax: `robot repair --input xref-need-of-repair.obo --annotation-property oboInOwl:hasDbXref --output results/xref-repaired.obo`

### CODE REVIEW

**Comment by balhoff** [comment]
This PR addresses @cmungall's request:

> add an option to migrate selected (or ALL) annotation assertions

However I didn't implement the ALL option. I don't think we would want to accidentally migrate `owl:deprecated` or `rdfs:label`. I suppose ALL could exclude just those two, but I think it is less magical to list the desired properties. But I am open to changing course.

**Comment by jamesaoverton** [comment]
@balhoff Thanks for including automated tests. ROBOT will run the integration tests during `mvn verify` with the example code from `repair.md` and check the example files, so it should not be necessary to have the same files tested as a unit test in `RepairOperationTest.java`.

When just comparing inputs to outputs, I prefer just the integration tests. When comparing different method calls, just the unit tests are better. If that's not clear, we can add better documentation to `CONTRIBUTING.md`.

**Comment by jamesaoverton** [comment]
@rctauber suggests that using an option map and resolving the annotation properties inside the operation will make life easier going forward, as more options are added to the command.

File: robot-core/src/main/java/org/obolibrary/robot/RepairOperation.java [hunk 1]
```diff
@@ -52,6 +48,22 @@ public static void repair(
     repair(ontology, ioHelper, getDefaultOptions(), mergeAxiomAnnotations);
   }
 
+  /**
+   * Repairs ontology
+   *
+   * @param ontology the OWLOntology to reapir
```
Comment (cmungall) [inline robot-core/src/main/java/org/obolibrary/robot/RepairOperation.java]: Typo!

**Comment by balhoff** [comment]
@jamesaoverton @rctauber can you say more about what you mean by "option map"? Is there another command that works this way which I can look at?

Oh—I think you mean passing a Map structure to the Operation.

**Comment by jamesaoverton** [comment]
Yes, I think Chris pioneered the use of `Map<String,String>` structures to pass options from a Command to an Operation. One benefit is avoiding the need to update the method signatures every time. Here's an example: https://github.com/ontodev/robot/blob/master/robot-core/src/main/java/org/obolibrary/robot/ReasonOperation.java#L43

**Comment by balhoff** [comment]
@jamesaoverton the method signatures in `RepairOperation` do already have a parameter for an option map. I looked at accepting the list of annotation properties this way, but it bothers me to mix command-line parsing into the operation (this is done in my PR in the command class using `CommandLineHelper`). I like the distinction in the ROBOT readme: "They should not contain IO or CLI code." So if you think method signatures have proliferated too much (I left the previous versions to not break binary compatibility) I can adapt, but I'm inclined to leave it this way.

**Comment by jamesaoverton** [comment]
@balhoff Ok, I'm fine with that. My only remaining comment is that the `properties` set is constructed but no used in `RepairCommand.java`.

**Comment by balhoff** [comment]
> @balhoff Ok, I'm fine with that. My only remaining comment is that the `properties` set is constructed but no used in `RepairCommand.java`.

Thanks for catching this! 😬 I fixed that, and also another issue I discovered with `mergeAxiomAnnotations` not being read correctly. Sorry for the long delay; this should be ready to review/merge.

**Comment by jamesaoverton** [comment]
Thanks @balhoff!

**Pull request merged**